runhouse/resources/hardware/cluster_factory.py

```diff
@@ -192,7 +194,7 @@ def cluster(
     )
     c.set_connection_defaults(**kwargs)
 
-    if den_auth:
+    if den_auth or rh.configs.telemetry_collection_enabled():
```

Comment:
do we need this here? if den auth is not enabled then we shouldn't be saving regardless